pkg/apis/apps/validation/validation.go

```diff
@@ -39,10 +39,10 @@ import (
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
 func ValidateStatefulSetName(name string, prefix bool) []string {
-	// TODO: Validate that there's name for the suffix inserted by the pods.
+	// TODO: Validate that there's room for the suffix inserted by the pods.
 	// Currently this is just "-index". In the future we may allow a user
 	// specified list of suffixes and we need  to validate the longest one.
-	return apimachineryvalidation.NameIsDNSSubdomain(name, prefix)
+	return apimachineryvalidation.NameIsDNSLabel(name, prefix)
```

Comment:
can we add an unit test https://github.com/kubernetes/kubernetes/blob/5bed61f7fd95b5b2570cafe05b550a88df10b4a2/pkg/apis/apps/validation/validation_test.go#L45 ?

Reply:
good call, added